pkg/operator/configobservation/apiserver/observer_apiserver_test.go

```diff
@@ -0,0 +1,415 @@
+package apiserver
+
+import (
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/tools/cache"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+func TestObserveUserClientCABundle(t *testing.T) {
+
+	existingConfig := map[string]interface{}{}
+
+	testCases := []struct {
```

Comment:
add an initial state here so you can test the transitions to new levels.

Reply:
No state managed as only configmaps are synced on observation.